WebApplication/MessageBroker.cs

```diff
@@ -2,27 +2,74 @@
 using System.Linq;
 using RabbitMQ.Client;
 using System.Text;
+using System.Threading;
+using System.Collections.Generic;
+using RabbitMQ.Client.Events;
+using System.IO;
+using RabbitMQ.Client.Exceptions;
 
-class EmitLogTopic
+namespace WebApplication
 {
-    public static string Send(Serilog.ILogger logger)
+    struct Message
     {
-        var factory = new ConnectionFactory() { HostName = "localhost" };
-        using (var connection = factory.CreateConnection())
-        using (var channel = connection.CreateModel())
+        public string routingKey;
+        public byte[] messageBody;
+        public Message(string routingKey, byte[] messageBody)
         {
-            channel.ExchangeDeclare(exchange: "topic_logs",
-                                    type: "topic");
+            this.routingKey = routingKey;
+            this.messageBody = messageBody;
+        }
+    }
+
+    interface IMessageBroker
+    {
+        void Send(string exchange, string routingKey, byte[] messageBody);
+        IList<Message> ReceiveNoWait(string queueName, int timeOutSeconds);
+    }
 
-            var routingKey = "anonymous.info";
-            var message = "Hello World!";
-            var body = Encoding.UTF8.GetBytes(message);
-            channel.BasicPublish(exchange: "topic_logs",
-                                 routingKey: routingKey,
-                                 basicProperties: null,
-                                 body: body);
-            logger.Information("[x] Sent " + routingKey + ":" + message);
-            return " [x] Sent " + routingKey + ":" + message;
+    class TopicExchange : IMessageBroker
```

Comment:
Incorrect naming. If you inherit from interface or class - new class must have part of interface/base class name